### PR TITLE
Fix hangs w/ bag.count/sum with distributed scheduler

### LINKED ISSUE #248: dask.bag.count hangs on distributed scheduler.
Whenever a dask.bag.count item is computed with the distributed scheduler it hangs.

### PR DESCRIPTION
**Pull request by cowlicks**
closes #248 

The main fix is 017241f basicall dask.core.flatten would do this
```python
>>> flatten('foo')
['f', 'o', 'o']
```
Now it does this
```python
>>>flatten('foo')
['foo']
```

I renamed `test_ipython_utils` to `test_distributed` and made it the place for tests that require a dask.distributed cluster to be running.

### CODE REVIEW

File: dask/core.py [hunk 1]
```diff
@@ -234,12 +234,15 @@ def flatten(seq):
     >>> list(flatten((1, 2, [3, 4]))) # support heterogeneous
     [1, 2, 3, 4]
     """
```
Comment (mrocklin) [inline dask/core.py]: test?

**Comment by mrocklin** [comment]
Awesome.  Can't tell you how satisfying it is to see a difficult to track down bug solved by someone else ;)

Any thoughts on how can we improve the debugging experience?

**Comment by cowlicks** [comment]
Hmm my debug experience was basically putting `log` calls everywhere until I surrounded the hanging line.

Being able to restart the `ipython` parallel cluster more quickly would be nice. Restarting the cluster takes just long enough for me to get distracted. So often I'd restart it, then get sidetracked while waiting.

 Or maybe instead we could just have some script for a local debugging cluster where we manually launch processes. This might be easy if we just targeted linux.

**Comment by cowlicks** [comment]
I added newlines in the log calls, this made them a lot more readable to me. It would be nice if we had aliases like `worker-1` instead of `tcp://....:6534 ` in the logs too.

File: dask/core.py (hunk 1 above)
Comment (cowlicks) [inline dask/core.py]: added

**Pull request merged**